src/Avalonia.Controls/Flyouts/MenuFlyoutPresenter.cs

```diff
@@ -47,5 +47,15 @@ protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e
                 }
             }
         }
+
+        protected internal override void PrepareContainerForItemOverride(Control element, object? item, int index)
+        {
+            base.PrepareContainerForItemOverride(element, item, index);
+
+            // Child menu items should not inherit the menu's ItemContainerTheme as that is specific
+            // for top-level menu items.
+            if ((element as MenuItem)?.ItemContainerTheme == ItemContainerTheme)
+                element.ClearValue(ItemContainerThemeProperty);
```

Comment:
Not sure if I understand.
When I set `<MenuFlyout ItemContainerTheme="" />` I would expect same container theme to be applied on flyout items **and** subitems.

Reply:
I agree with you. But I have decided to respect the same behavior of the menus.

https://github.com/AvaloniaUI/Avalonia/blob/61916d7e3f3f4f7966d5702d22a1819b3cb26ad3/src/Avalonia.Controls/Menu.cs#L97-L105

I've opened a discussion #10060 about it, but no one has responded yet.

Reply:
I think it's different in case of Menu control. 
As in Menu control you have top level menu items that are always rendered horizontally and flyout/popup menus with a different styles. And there you wouldn't expect same style to be applied everywhere.

Reply:
I personally expected the same behavior also for the Menu as it happens in WPF. I'd like to do a PR that changes the behavior of the menus, but I'm waiting for your response first. I don't want invest   time unnecessarily.